File: app/src/main/java/com/denizd/substitutionplan/FoodAdapter.java
package com.denizd.substitutionplan;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class FoodAdapter extends RecyclerView.Adapter<FoodAdapter.CardViewHolder> {
    private List<Food> mFood;

    public static class CardViewHolder extends RecyclerView.ViewHolder {
        public TextView mFood;

        public CardViewHolder(@NonNull View itemView) {
            super(itemView);
            mFood = itemView.findViewById(R.id.cardInfoText);
        }
    }

    public FoodAdapter(ArrayList<Food> food) {
        mFood = food;
    }

    @NonNull
    @Override
    public CardViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.just_a_card, parent, false);
        CardViewHolder cvh = new CardViewHolder(v);
        return cvh;
    }

    public Food getFoodAt(int i) {
        return mFood.get(i);
    }

    @Override
    public void onBindViewHolder(@NonNull CardViewHolder holder, int position) {
        Food currentItem = mFood.get(position);

        holder.mFood.setText(currentItem.getFood());
    }

    @Override
    public int getItemCount() {
        return mFood.size();
    }

    public void setFood(List<Food> food) {
        mFood = food;
        notifyDataSetChanged();
    }
}

File: app/src/main/java/com/denizd/substitutionplan/FoodAdapter.kt
package com.denizd.substitutionplan

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import java.util.ArrayList
import androidx.recyclerview.widget.RecyclerView

class FoodAdapter(food: ArrayList<Food>) : RecyclerView.Adapter<FoodAdapter.CardViewHolder>() {
    private var mFood: List<Food>? = null

    class CardViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var mFood: TextView

        init {
            mFood = itemView.findViewById(R.id.cardInfoText)
        }
    }

    init {
        mFood = food
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CardViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.just_a_card, parent, false)
        return CardViewHolder(v)
    }

    fun getFoodAt(i: Int): Food {
        return mFood!![i]
    }

    override fun onBindViewHolder(holder: CardViewHolder, position: Int) {
        val currentItem = mFood!![position]

        holder.mFood.text = currentItem.food
    }

    override fun getItemCount(): Int {
        return mFood!!.size
    }

    fun setFood(food: List<Food>) {
        mFood = food
        notifyDataSetChanged()
    }
}
